lib/puppet/provider/archive/ruby.rb

```diff
@@ -1,13 +1,5 @@
-begin
-  require 'puppet_x/bodeco/archive'
-  require 'puppet_x/bodeco/util'
-rescue LoadError
-  require 'pathname' # WORK_AROUND #14073 and #7788
-  archive = Puppet::Module.find('archive', Puppet[:environment].to_s)
-  raise(LoadError, "Unable to find archive module in modulepath #{Puppet[:basemodulepath] || Puppet[:modulepath]}") unless archive
-  require File.join archive.path, 'lib/puppet_x/bodeco/archive'
-  require File.join archive.path, 'lib/puppet_x/bodeco/util'
-end
+require File.dirname(__FILE__) + '/../../../puppet_x/bodeco/archive.rb'
```

Comment:
Not sure about this.  The code you're removing looks like it was specifically added as a workaround for https://tickets.puppetlabs.com/browse/SERVER-973

https://github.com/voxpupuli/puppet-corosync/blob/master/lib/puppet/provider/cs_clone/crm.rb#L4 is identical?  But in that module the comments also reference SERVER-973

ping @roidelapluie - Can you clarify?